### PR TITLE
fix: cookie expiration date

### PR DESCRIPTION
**Pull request by ntheile**
This PR fixes the cookie session expiration date. Before it would expire after a few hours. Now it will not expire until the kratos session expires (kratos session lifespan setting, typically 30 days/720h)

### CODE REVIEW

File: src/servers/middlewares/auth-router.ts [hunk 1]
```diff
@@ -168,21 +169,36 @@ authRouter.post(
...
+            sameSite: "none",
+            secure: kratosSessionCookie.secure,
+            httpOnly: kratosSessionCookie.httpOnly,
+            path: kratosSessionCookie.path,
+          },
+        )
+        const session = await kratosPublic.toSession({ cookie: kratosCookieStr })
+        const expiresAt = session.data.expires_at
+          ? new Date(session.data.expires_at)
+          : new Date(new Date().setDate(new Date().getDate() + 30))
```
Comment (nicolasburtey) [inline src/servers/middlewares/auth-router.ts:187]: should avoid the magic number, or document it with a dedicated const

**Comment by nicolasburtey** [comment]
> This PR fixes the cookie session expiration date. Before it would expire after a few hours. Now it will not expire until the kratos session expires (kratos session lifespan setting, typically 30 days/720h)

the idea of shorter session for cookie (versus token) actually make sense for me? but I understand this should be a setting from kratos not from our code ideally.

but I don't think it's possible from kratos to say that cookie should have short duration than token

File: src/servers/middlewares/auth-router.ts (hunk 1 above)
Comment (ntheile) [reply src/servers/middlewares/auth-router.ts:187]: yeah chatgpt wrote that for me lol. it stands for 30 days from now. where should i put the dedicated const?

**Comment by ntheile** [comment]
> 

yeah it was a bug, the maxAge is used in browsers and the expires field in cookies are deprecated now https://mrcoles.com/blog/cookies-max-age-vs-expires/#:~:text=Quick%20Answer%3A,mostly)%20all%20browsers%20support%20expires

**Review by bodymindarts — APPROVED** [review]

**Pull request merged**